uint32 counter
float64 distance
bool manipulatorIsClose
bool activateAdjuster
